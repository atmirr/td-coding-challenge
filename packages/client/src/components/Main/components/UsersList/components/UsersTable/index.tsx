import React from 'react';
import {
  Paper,
  TableContainer,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
} from '@material-ui/core';
import UserRow from './components/UserRow';
import { User as UserType } from 'src/components/Main/components/UsersList';

function UsersTable({ items }: { items: UserType[] }) {
  return (
    <TableContainer data-testid="users-table" component={Paper}>
      <Table aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>ID</TableCell>
            <TableCell>Name</TableCell>
            <TableCell>Username</TableCell>
            <TableCell align="center">Actions</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {items.map(({ id, name, username }: UserType) => (
            <UserRow key={id} id={id} name={name} username={username} />
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default UsersTable;
