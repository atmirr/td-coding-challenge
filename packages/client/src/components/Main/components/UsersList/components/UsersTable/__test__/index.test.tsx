import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';
import { screen, render } from '@testing-library/react';
import UsersTable from '../index';

const props = {
  items: [
    {
      id: 1,
      name: 'Sunny Wucker',
      username: 'madisyn.wuckert51',
    },
    {
      id: 2,
      name: 'Indigo Shady',
      username: 'hilda_turcotte',
    },
  ],
};

describe('<UsersTable />', () => {
  it('should renders the component', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <UsersTable {...props} />
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should display all the users' info which passed by items prop", () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <UsersTable {...props} />
      </MemoryRouter>,
    );
    props.items.forEach((user, index) => {
      expect(screen.queryAllByText(user.id).length).toBe(1);
      expect(screen.queryAllByText(user.name).length).toBe(1);
      expect(screen.queryAllByText(user.username).length).toBe(1);
      const detailButtons = container.querySelectorAll('a');
      expect(detailButtons[index]).toHaveAttribute('href', `/user/${user.id}`);
    });
  });

  it('should display all the column labels', () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <UsersTable {...props} />
      </MemoryRouter>,
    );
    expect(screen.queryAllByText(/ID/).length).toBe(1);
    expect(screen.queryAllByText(/Name/).length).toBe(1);
    expect(screen.queryAllByText(/Username/).length).toBe(1);
    expect(screen.queryAllByText(/Actions/).length).toBe(1);
  });
});
