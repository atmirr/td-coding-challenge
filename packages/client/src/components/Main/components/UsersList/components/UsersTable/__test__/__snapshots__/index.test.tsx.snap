// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersTable /> should renders the component 1`] = `
<div
  class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="users-table"
>
  <table
    aria-label="simple table"
    class="MuiTable-root"
  >
    <thead
      class="MuiTableHead-root"
    >
      <tr
        class="MuiTableRow-root MuiTableRow-head"
      >
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          ID
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Username
        </th>
        <th
          class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
          scope="col"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          1
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Sunny Wucker
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          madisyn.wuckert51
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <a
            href="/user/1"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-detailIcon-1"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          2
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          Indigo Shady
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
          hilda_turcotte
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
        >
          <a
            href="/user/2"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-detailIcon-1"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
            </svg>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
